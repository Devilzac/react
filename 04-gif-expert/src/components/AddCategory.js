import React, {useState} from 'react';


export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState("Hola mundo");

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        const text = inputValue.trim();
        if(text.length <= 1) return;
       //setCategories(cats => [...cats,inputValue]);
       onNewCategory(text);
       setInputValue('');
    }

  return (
    <form onSubmit={handleSubmit}>
        <input 
            type="text"
            value={inputValue}
            onChange={handleInputChange}
        />
    </form>
  )
}
