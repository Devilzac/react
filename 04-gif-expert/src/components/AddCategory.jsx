import React, {useState} from 'react';
import PropTypes from 'prop-types'; 

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState("Hola mundo");

    const handleInputChange = (e)=>{
        setInputValue(e.target.value);
    }

    const handleSubmit = (e)=>{
        e.preventDefault();
        const text = inputValue.trim();
        if(text.length <= 1) return;
       //setCategories(cats => [...cats,inputValue]);
       onNewCategory(text);
       setInputValue('');
    }

  return (
    <form onSubmit={handleSubmit}>
        <input 
            type="text"
            value={inputValue}
            onChange={handleInputChange}
        />
    </form>
  )
}


AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}