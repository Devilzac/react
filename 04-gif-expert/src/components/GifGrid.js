import React from 'react'

export default function GifGrid({category}) {
const gifs =[1,2,3,4,5,6]

  return (
    <>
        <h3>{ category }</h3>
       {
        gifs.map(gif =>(
            <p key={gif}>gif</p>
            ))
       }
    </>
  )
}
