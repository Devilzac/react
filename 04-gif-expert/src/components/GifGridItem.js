import React from 'react'

export const GifGridItem = ({title, url,id}) => {
  return (
    <div className="card">
        <img width="400" height="200" src={url} alt={title}/>  
        <p>{title}</p>
    </div>

  )
}
