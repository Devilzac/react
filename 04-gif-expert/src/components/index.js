export * from './AddCategory';
export * from './GifGrid';
export * from './GifExpertApp';
export * from './GifGridItem';