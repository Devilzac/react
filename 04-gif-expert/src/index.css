* {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
}

.card-grid{
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
}
.card{
  align-content: center;
  align-items: center;
  background-color: white;
  border-radius: 10px;
  box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.3);
  display: flex;
  flex-direction: column;
  height: 0%;
  justify-content: center;
  margin: 10px;
  
}

.card-grid img{
  object-fit: cover;
  width: 400px;
  height: 200px;
}