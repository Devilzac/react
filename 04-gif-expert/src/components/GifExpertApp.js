import React, { useState } from 'react'
import { AddCategory } from './AddCategory';
import GifGrid from './GifGrid';

export const GifExpertApp = () => {
    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    const onAddCategory = (newCategory)=>{
        if(categories.includes(newCategory)) return;
        setCategories([ newCategory, ...categories]);
    }

  return (
    <>
        <AddCategory 
            onNewCategory={ (value) => onAddCategory(value) }
        />
        <hr />
        {
            categories.map((category) =>(
                    <GifGrid 
                        key={ category }
                        category={ category } 
                    />
                ))
        }
        
    </>
  )
}
