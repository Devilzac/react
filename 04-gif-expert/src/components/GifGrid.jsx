import React, { useEffect, useState } from 'react';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';
//import { getGifs } from '../helpers/getGifs';

export function GifGrid({category}) {

  const { images, isLoading} = useFetchGifs(category);
  console.log({images, isLoading, category});



  return (
    <>
      <h3>{ category }</h3>
      {
        isLoading && (<h1>cargando...</h1>) 
      }
        <div className='card-grid'>
          {images.map( (images) =>(
            <GifGridItem key={images.id} 
              {...images}
            />
          ))}
        </div>
    </>
  )
}

GifGrid.propTypes= {
  category: PropTypes.string.isRequired,
}