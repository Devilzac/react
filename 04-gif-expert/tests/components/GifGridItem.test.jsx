import React from 'react';
import { render } from '@testing-library/react';
import {GifGridItem} from '../../src/components';

describe('Pruebas de <GifGridItem />', () => {

    const title = "dr strange";
    const url="google.com";
    const id= 1;
    test('debe de ser igual al snapshot', ()=>{

        const {container} = render(<GifGridItem title={title} url={url} id={id} />);              
        expect(container).toMatchSnapshot();
    });

});