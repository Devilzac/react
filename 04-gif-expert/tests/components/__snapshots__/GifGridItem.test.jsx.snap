// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <GifGridItem /> debe de ser igual al snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="dr strange"
      height="200"
      src="google.com"
      width="400"
    />
    <p>
      dr strange
    </p>
  </div>
</div>
`;
